borg/fuse.py

```diff
@@ -76,6 +76,16 @@ def process_archive(self, archive, prefix=[]):
             data = self.key.decrypt(key, chunk)
             unpacker.feed(data)
             for item in unpacker:
+                try:
+                    # This can happen when an archive was created with a command line like
+                    # $ borg create ... dir1/file dir1
+                    # In this case the code below will have created a default_dir inode for dir1 already.
+                    inode = self._find_inode(item[b'path'], prefix)
+                except KeyError:
+                    pass
+                else:
+                    self.items[inode] = item
```

Comment:
In line 108 the inode has been either assigned in the first if branch (hardlink) from an existing inode (ok). Or it came from the other branch for "all other items" (including, usually, directories, except if it was added via the default_dir code path below that for a precursor segment), so it's a cached inode where the inode is the offset'ed index into the items cache file.